libraries/discriminator/src/discriminator.rs

```diff
@@ -0,0 +1,72 @@
+//! The traits and types used to create a discriminator for a type
+
+use {
+    bytemuck::{Pod, Zeroable},
+    solana_program::{hash, program_error::ProgramError},
+};
+
+/// A trait for managing 8-byte discriminators in a slab of bytes
+pub trait SplDiscriminates {
```

Comment:
nit: sorry, I'm leaning more towards `SplDiscriminate` in the end. I can't find any examples in the rust std lib using a non-imperative verb.

Reply:
😭

Reply:
Please feel free to convince me otherwise!